### PR TITLE
Feat/serialize deserialize handler

### LINKED ISSUE #238: [getLambdaHandler] improve lambda handler's output typing with `outputSchema: undefined`
When `outputSchema` is set to `undefined` in a contract, lambda handler's output typing inferred by `getLambdaHandler` is `Promise<undefined>`.

```
export type OutputType<Contract extends ApiGatewayContract> =
  Contract['outputSchema'] extends JSONSchema
    ? FromSchema<Contract['outputSchema']>
    : undefined;
```

This causes a TypeScript error when creating a lambda that does not return anything because TypeScript expects a `Promise<undefined>` instead of a `Promise<void>`.

Setting `outputSchema` in the contract to `undefined` seems to be the proper configuration when a lambda does not return anything. Typing lambda handler's output as `Promise<void>` would solve this.

```
export type OutputType<Contract extends ApiGatewayContract> =
  Contract['outputSchema'] extends JSONSchema
    ? FromSchema<Contract['outputSchema']>
    : void;
```

### PR DESCRIPTION
**Pull request by Sc0ra**
Fixes #238

### CODE REVIEW

File: packages/serverless-contracts/src/contracts/apiGateway/features/httpLambdaHandler.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  statusCode: 200,
+  body: JSON.stringify(handlerResponse),
+});
+
+const ajv = new Ajv();
+
+export const getHttpLambdaHandler =
+  <Contract extends ApiGatewayContract>(contract: Contract) =>
+  (
+    handler: HandlerType<typeof contract>,
```
Comment (fargito) [inline packages/serverless-contracts/src/contracts/apiGateway/features/httpLambdaHandler.ts:45]:
```suggestion
    handler: HandlerType<Contract>,
```

File: packages/serverless-contracts/src/contracts/apiGateway/features/httpLambdaHandler.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  ): CompleteHandlerType<typeof contract> =>
```
Comment (fargito) [inline packages/serverless-contracts/src/contracts/apiGateway/features/httpLambdaHandler.ts:46]:
```suggestion
  ): CompleteHandlerType<Contract> =>
```

File: packages/serverless-contracts/src/contracts/apiGateway/features/httpLambdaHandler.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+};
+
+const handlerResponseToLambdaResult = <Contract extends ApiGatewayContract>(
+  handlerResponse: OutputType<Contract>,
+): LambdaReturnType<Contract> => ({
+  statusCode: 200,
+  body: JSON.stringify(handlerResponse),
+});
+
+const ajv = new Ajv();
```
Comment (fargito) [inline packages/serverless-contracts/src/contracts/apiGateway/features/httpLambdaHandler.ts:40]: Shouldn't we move this inside the function body?

File: packages/serverless-contracts/src/contracts/apiGateway/types/lambdaHandler.ts [hunk 4]
```diff
@@ -14,7 +18,7 @@ import {
   OutputType,
 } from './common';
 import { InputSchemaType } from './input';
-import type { DefinedProperties } from './utils';
+import { DefinedProperties } from './utils';
```
Comment (fargito) [inline packages/serverless-contracts/src/contracts/apiGateway/types/lambdaHandler.ts:21]: Why is this necessary?

Comment (Sc0ra) [reply packages/serverless-contracts/src/contracts/apiGateway/types/lambdaHandler.ts:21]: Don't know why i had to add it in the first place, but it works without it now

File: packages/serverless-contracts/src/contracts/apiGateway/types/lambdaHandler.ts [hunk 5]
```diff
@@ -52,5 +56,26 @@ export type HandlerType<Contract extends ApiGatewayContract> = (
...
+    ? APIGatewayProxyEventBase<AuthorizerContext<Contract['authorizerType']>>
+    : APIGatewayProxyEventV2WithRequestContext<
+        APIGatewayEventRequestContextV2WithAuthorizer<
+          AuthorizerContext<Contract['authorizerType']>
+        >
+      >;
+
+export type LambdaReturnType<Contract extends ApiGatewayContract> =
+  Contract['integrationType'] extends 'restApi'
+    ? APIGatewayProxyResult
```
[the remark this answers is not in the document]
Comment (Sc0ra) [reply packages/serverless-contracts/src/contracts/apiGateway/types/lambdaHandler.ts:76]: It does not, for api gateway v1, the proxy result format is not generic

File: packages/serverless-contracts/src/contracts/apiGateway/features/httpLambdaHandler.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+export const getHttpLambdaHandler =
+  <Contract extends ApiGatewayContract>(contract: Contract) =>
+  (
+    handler: HandlerType<typeof contract>,
+  ): CompleteHandlerType<typeof contract> =>
+  async event => {
+    try {
+      const parsedEvent = proxyEventToHandlerEvent<Contract>(event);
+
+      if (contract.bodySchema !== undefined) {
```
[the remark this answers is not in the document]
Comment (Sc0ra) [reply packages/serverless-contracts/src/contracts/apiGateway/features/httpLambdaHandler.ts:49]: Maybe it can be a next feature, but for the moment, i wanted to implement what already exists with the middlewares we already use.
But indeed, we have the ability to validate other things than the body on input data

File: backend/orchestrator/functions/requestSyncDeployment/handler.ts [hunk 7]
```diff
@@ -1,12 +1,11 @@
...
 
 import { requestSyncDeployment } from '@swarmion/orchestrator-contracts';
-import { getLambdaHandler } from '@swarmion/serverless-contracts';
-import { applyHttpMiddlewares } from '@swarmion/serverless-helpers';
+import { getHttpLambdaHandler } from '@swarmion/serverless-contracts';
 
 import ServiceEventEntity from 'libs/dynamodb/models/serviceEvent';
 
-const handler = getLambdaHandler(requestSyncDeployment)(async event => {
+export const main = getHttpLambdaHandler(requestSyncDeployment)(async event => {
```
Comment (adriencaccia) [inline backend/orchestrator/functions/requestSyncDeployment/handler.ts:8]: We lose the validation of the `inputSchema` that we add with `applyHttpMiddlewares` + `inputSchema` here though

Comment (adriencaccia) [reply backend/orchestrator/functions/requestSyncDeployment/handler.ts:8]: Gottem haha
It was done in the upcoming commits 😮‍💨

**Comment by adriencaccia** [comment]
Ok for me @Sc0ra 🚀 
As we discussed, only the documentation left to do

File: user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md [hunk 8]
```diff
@@ -166,6 +166,22 @@ const handler = getLambdaHandler(myContract)(async event => {
 });
 ```
 
+If you use your lambda as an ApiGateway integration, you will need to use middlewares to parse the string body, validate the input and output formats and serialize the output body. You may also need to handler error cases.
```
Comment (adriencaccia) [inline user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md:169]: We can add this in a dedicated section like

```ts
### Validate the input and output at runtime
```

**Review by adriencaccia — CHANGES_REQUESTED** [review]

File: user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md (hunk 8 above)
Comment (Sc0ra) [reply user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md]: Documentation text updated, in a new section

**Comment by Sc0ra** [comment]
Should also fix #238 with the last commit

**Review by adriencaccia — APPROVED** [review]

**Comment by adriencaccia** [comment]
LGTM @Sc0ra you can merge 😉
